xds/internal/client/client_watchers_service.go

```diff
@@ -27,7 +27,13 @@ import (
 type ServiceUpdate struct {
 	// WeightedCluster is a map from cluster names (CDS resource to watch) to
 	// their weights.
+	//
+	// This field is only set when routing is disabled (not enabled) by env
```

Comment:
Similar comment as in the `rdsUpdate` struct. This struct could simply contain a slice of routes, and if required a boolean indicating whether to use routing balancer.